src/pmproxy/src/server.c

```diff
@@ -60,6 +60,42 @@ proxylog(pmLogLevel level, sds message, void *arg)
     pmNotifyErr(priority, "%s%s", state, message);
 }
 
+static void
+refresh_proxy_mem_metrics(struct proxy *proxy)
+{
+    pmAtomValue		*value;
+    uint64_t		vsz, rss;
+    int			fd;
+    char		buf[MAXPATHLEN];
+    static		pid_t pid = 0;
+    static		int pagesz = 0;
+
+    if (!pid)
+    	pid = getpid();
+    sprintf(buf, "/proc/%u/statm", pid);
```

Comment:
This is Linux-specific so I guess this whole function should be conditionally compiled depending on platform.